contrib/platform/test/com/sun/jna/platform/win32/WevtapiTest.java

```diff
@@ -0,0 +1,574 @@
+                for (int i = 0; i < returned.getValue(); i++) {
+                    read++;
+                    buff = WevtapiUtil.EvtRender(contextHandle, eventArray[i], Winevt.EVT_RENDER_FLAGS.EvtRenderEventValues, propertyCount);
+                    assertThat("PropertyCount", propertyCount.getValue(), is(2));
+                    useMemory(evtVariant, buff, 0);
+                    assertThat("Security#UserID", ((WinNT.PSID) evtVariant.getValue()).getSidString(), is("S-1-5-21-3178902164-3053647283-518304804-1001"));
+                    useMemory(evtVariant, buff, 1);
+                    assertThat("Provider#GUID", ((Guid.GUID) evtVariant.getValue()).toGuidString(), is("{B0AA8734-56F7-41CC-B2F4-DE228E98B946}"));
+                }
+            }
+
+            assertThat(read, is(1));
+        } finally {
+            // test EvtClose
+            if (queryHandle != null) {
+                Wevtapi.INSTANCE.EvtClose(queryHandle);
+            }
+            if (contextHandle != null) {
+                Wevtapi.INSTANCE.EvtClose(contextHandle);
+            }
+        }
+    }
+
+    private void useMemory(Winevt.EVT_VARIANT evtVariant, Memory buff, int index) {
+        evtVariant.use(buff.share(evtVariant.size() * index));
+        evtVariant.read();
+    }
+
+
+    public void testEvtOpenLog() throws Exception {
+        File testEvtx = new File(getClass().getResource("/res/WevtapiTest.sample1.evtx").toURI());
+        EVT_HANDLE logHandle = Wevtapi.INSTANCE.EvtOpenLog(null, testEvtx.getAbsolutePath(),
+                Winevt.EVT_OPEN_LOG_FLAGS.EvtOpenFilePath);
+        if (logHandle == null) {
+            throw new Win32Exception(Kernel32.INSTANCE.GetLastError());
+        }
+        Memory buff = new Memory(1024);
+        IntByReference buffUsed = new IntByReference();
+        if (!Wevtapi.INSTANCE.EvtGetLogInfo(logHandle, Winevt.EVT_LOG_PROPERTY_ID.EvtLogFileSize, (int) buff.size(), buff, buffUsed)) {
+            if (Kernel32.INSTANCE.GetLastError() == WinError.ERROR_INSUFFICIENT_BUFFER) {
+                buff = new Memory(buffUsed.getValue());
+                if (!Wevtapi.INSTANCE.EvtGetLogInfo(logHandle, Winevt.EVT_LOG_PROPERTY_ID.EvtLogFileSize, (int) buff.size(), buff, buffUsed)) {
+                    throw new Win32Exception(Kernel32.INSTANCE.GetLastError());
+                }
+            } else {
+                throw new Win32Exception(Kernel32.INSTANCE.GetLastError());
+            }
+        }
+        assertThat(buff.getLong(0), is(69632L));
+    }
+
+
+    public void testEvtOpenChannelEnum() throws Exception {
+
+        EVT_HANDLE channelHandle = null;
+        List<String> channelList = new ArrayList<String>();
+        try {
+            channelHandle = Wevtapi.INSTANCE.EvtOpenChannelEnum(null, 0);
+            if (channelHandle == null) {
+                throw new Win32Exception(Kernel32.INSTANCE.GetLastError());
+            }
+
+            Memory buff = new Memory(1024);

```

Comment:
This needs to be adjusted if you switch to char[] in Wevtapi. See 245 for the reason.